components/EditForm/index.js

```diff
@@ -0,0 +1,85 @@
+import { v4 as uuidv4 } from "uuid";
+import { useImmer } from "use-immer";
+import useStore from "../../hooks/useStore";
+
+export default function EditForm({ project, toggleEdit }) {
+  const [projectName, setProjectName] = useImmer(project.name);
+  const [items, setItems] = useImmer(project.items);
+  const [editProject, deleteProject] = useStore((state) => [
+    state.editProject,
+    state.deleteProject,
+  ]);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const editedProject = {
+      name: event.target.projectname.value,
+      items: items,
+      id: project.id,
+    };
+    editProject(editedProject);
+    toggleEdit();
+  }
+  //add a new item
+  const handleAddItem = () => {
+    setItems([...items, { value: "", name: "", price: "", id: uuidv4() }]);
+  };
+
+  // values change
+  const handleItemChange = (index, field, value) => {
+    setItems((draft) => {
+      draft[index][field] = value;
+    });
+  };
+
+  //change project-name
+  const handleProjectNameChange = (event) => {
+    setProjectName(event.target.value);
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        type="text"
+        name="projectname"
+        placeholder="project name"
+        value={projectName}
+        onChange={handleProjectNameChange}
+      />
```

Comment:
`required` might be useful for the project name

Reply:
i will add verification in an other user story